edn/src/types.rs

```diff
@@ -24,7 +26,7 @@ pub enum Value {
     Float(OrderedFloat<f64>),
     Text(String),
     Symbol(String),
-    Keyword(String),
+    Keyword(keyword::Keyword),
```

Comment:
No, `namespaced` is the specific constructor function for `:foo/bar` as opposed to `:bar`.

Keywords are essentially in two disjoint sets: those with namespaces and those without.

I modeled this as an `Option<String>` for the namespace, because I didn't want yet another layer of enums. I modeled this as two constructor functions with one helper (`perhaps_namespaced`) to avoid passing `None` in places.

This *is* a spot where you can use some trait tricks to allow coercion from `&str` to `Some<&str>` to `Some<String>` within the same signature, but that would make the constructors relatively more complicated.